tests/functional/testUtapiClient.js

```diff
@@ -1,37 +1,90 @@
 import assert from 'assert';
-import { mapSeries, series } from 'async';
+import { map, series } from 'async';
 import UtapiClient from '../../src/lib/UtapiClient';
-import MemoryBackend from '../../src/lib/backend/Memory';
 import Datastore from '../../src/lib/Datastore';
-import { getBucketCounters, getMetricFromKey } from '../../src/lib/schema';
+import redisClient from '../../src/utils/redisClient';
+import { Logger } from 'werelogs';
+import { getBucketCounters, getMetricFromStateKey,
+    getBucketStateKeys } from '../../src/lib/schema';
 const testBucket = 'foo';
-const memBackend = new MemoryBackend();
 const datastore = new Datastore();
 const utapiClient = new UtapiClient();
 const reqUid = 'foo';
-datastore.setClient(memBackend);
+const redis = redisClient({ host: '127.0.0.1', port: 6379 }, Logger);
+const putOperations = ['PutObject', 'CopyObject', 'UploadPart',
+    'CompleteMultipartUpload'];
+
+datastore.setClient(redis);
 utapiClient.setDataStore(datastore);
 
-function _assertCounters(bucket, cb) {
+function _assertCounters(bucket, valueObj, cb) {
     const counters = getBucketCounters(testBucket);
-    return mapSeries(counters, (item, next) =>
-        memBackend.get(item, (err, res) => {
+    return map(counters, (item, next) =>
```

Comment:
minor: `next` could be called simply `cb` or `done`, because calls are neither ordered nor serialized (anymore).